### PR TITLE
Reduce duplication for maintaining docs of undesirable functions/operators

### LINKED ISSUE #1464: Is it possible to avoid repetition across code and docs for undesirable linters?
Currently, we are repeating the same text twice:

Once in docs-

https://github.com/r-lib/lintr/blob/1a567f09cee74f5eb264a19dfde78f68d8e82278/R/zzz.R#L51

The other time in code-

https://github.com/r-lib/lintr/blob/1a567f09cee74f5eb264a19dfde78f68d8e82278/R/zzz.R#L101-L102

This means that we run the risk of forgetting to update one if the other is updated.

Can we think of a way to avoid this repetition?

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #1464

### CODE REVIEW

File: man/default_undesirable_functions.Rd [hunk 1]
```diff
@@ -24,54 +24,38 @@ Lists of function names and operators for \code{\link[=undesirable_function_lint
...
-change instead.
-\item \code{\link[=Sys.setlocale]{Sys.setlocale()}} permanently modifies the session locale. Use \code{\link[withr:with_locale]{withr::with_locale()}} for a temporary change
-instead.
-\item \code{\link[=trace]{trace()}} traps a function and causes execution of arbitrary code when that function is run. It should be removed.
-\item \code{\link[=undebug]{undebug()}} is only useful for interactive debugging with \code{\link[=debug]{debug()}}. It should be removed.
-\item \code{\link[=untrace]{untrace()}} is only useful for interactive debugging with \code{\link[=trace]{trace()}}. It should be removed.
+\item \code{\link[=attach]{attach()}} As an alternative, use roxygen2's @importFrom statement in packages, or \code{::} in scripts. \code{\link[=attach]{attach()}} modifies the global search path.
+\item \code{\link[=browser]{browser()}} As an alternative, remove this likely leftover from debugging. It pauses execution when run.
+\item \code{\link[=debug]{debug()}} As an alternative, remove this likely leftover from debugging. It traps a function and causes execution to pause when that function is run.
```
Comment (AshesITR) [inline man/default_undesirable_functions.Rd:33]: This reads clunky 😅
No need to fix here.

File: man/default_undesirable_functions.Rd [hunk 2]
```diff
@@ -24,54 +24,38 @@ Lists of function names and operators for \code{\link[=undesirable_function_lint
...
 The following operators are sometimes regarded as undesirable:
 \itemize{
-\item \code{\link[base:ns-dblcolon]{:::}} accesses non-exported functions inside packages. Code relying on these is
-likely to break in future versions of the package because the functions are not part of the public interface and
-may be changed or removed by the maintainers without notice.
-Use public functions via \code{::} instead.
-\item \code{\link[base:assignOps]{<<-}} and \verb{->>} assign outside the current environment in a way that can be hard to reason
-about. Prefer fully-encapsulated functions wherever possible, or, if necessary, assign to a specific environment
-with \code{\link[=assign]{assign()}}. Recall that you can create an environment at the desired scope with \code{\link[=new.env]{new.env()}}.
+\item \code{<<-} As an alternative, It assigns outside the current environment in a way that can be hard to reason about. Prefer fully-encapsulated functions wherever possible, or, if necessary, assign to a specific environment with assign(). Recall that you can create an environment at the desired scope with new.env().
```
Comment (AshesITR) [inline man/default_undesirable_functions.Rd:57]: That's a downgrade regarding links.

Comment (MichaelChirico) [reply man/default_undesirable_functions.Rd:57]: Yea, wasn't sure how much to invest here. Anything besides linking the operator itself you think it worth hacking a link onto?

File: R/zzz.R [hunk 3]
```diff
@@ -207,6 +167,28 @@ default_undesirable_functions <- all_undesirable_functions[names(all_undesirable
   NULL
 )]
 
+rd_auto_link <- function(x) {
+  x <- unlist(x)
+  x <- gsub("withr::([a-zA-Z0-9._]+)\\(\\)", "\\\\code{\\\\link[withr:\\1]{withr::\\1()}}", x)
```
Comment (AshesITR) [inline R/zzz.R:172]: Maybe capture the package in a group so other packages, if used in the future, are supported?

Comment (MichaelChirico) [reply R/zzz.R]: Hmm I intentionally didn't do that, for a reason I now forget. Maybe I've just improved the other regex enough to avoid overlap. Fixed.

**Review by AshesITR — APPROVED** [review]

**Pull request merged**